File: front/src/services/process.service.js
import ApiService from "./api.service";

export default class ProcessService extends ApiService {

    constructor() {
        super('/api/process');
    }

    saveProcess(process) {
        return this.post(process);
    }

    editProcess(process) {
        return this.post(process);
    }

    deleteById(id) {
        return this.delete(id);
    }

    findAll() {
        return this.get();
    }

    findById(id) {
        return this.get(`${id}`);
    }

    findAllUsersFinishers() {
        return this.get('users/finishers');
    }

}

File: backend/src/main/java/com/softplan/desafio/api/service/ProcessService.java
package com.softplan.desafio.api.service;

import com.softplan.desafio.api.model.Process;
import com.softplan.desafio.api.repository.ProcessRepository;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
public class ProcessService {

    private final ProcessRepository repository;

    public ProcessService(ProcessRepository repository) {
        this.repository = repository;
    }

    public List<Process> findAll() {
        return repository.findAll();
    }

    public Optional<Process> findById(Long id) {
        return repository.findById(id);
    }

    public Process save(Process process) {
        return repository.save(process);
    }

    public void deleteById(Long id) {
        repository.deleteById(id);
    }

}
